server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestClusterStateAction.java

```diff
@@ -138,13 +150,16 @@ public RestResponse buildResponse(final ClusterStateResponse response,
                                         builder.field(Fields.CLUSTER_NAME, response.getClusterName().value());
                                         ToXContent.Params params = new ToXContent.DelegatingMapParams(
                                                 singletonMap(Metadata.CONTEXT_MODE_PARAM, Metadata.CONTEXT_MODE_API), request);
-                                        response.getState().toXContent(builder, params);
+                                        final ClusterState responseState = response.getState();
+                                        if (responseState != null) {
```

Comment:
Only question pretty much on this one: why can this be `null` all of a sudden?

Reply:
It was always thus:

https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/admin/cluster/state/TransportClusterStateAction.java#L109

TBF this only happens if it times out waiting for a particular metadata version, which in practice means the client is CCR and therefore not going via the REST layer anyway.